File: deploy.py
import subprocess
import json
import sys

try:
    with open("deploy_env.json", "r", encoding="utf-8") as f:
        env_data = json.load(f)

    env_string = json.dumps(env_data)

    print("Deploying to InsForge via Python subprocess with shell=True...")
    # By running with shell=True on Windows, we execute through cmd.exe which respects wrapped double quotes
    result = subprocess.run(["npx", "@insforge/cli", "deployments", "deploy", "--env", env_string, "."], capture_output=True, text=True, shell=True)
    
    print("STDOUT:")
    print(result.stdout)
    print("STDERR:")
    print(result.stderr)
    
    if result.returncode != 0:
        raise Exception(f"Deployment process failed with exit code: {result.returncode}")
        
    print("Deployment completed successfully!")
except Exception as e:
    print("Deployment failed:", e)
    sys.exit(1)

File: deploy.js
const { execFileSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const envData = JSON.parse(fs.readFileSync('deploy_env.json', 'utf8'));
const envString = JSON.stringify(envData);

// Resolve npx entry point relative to node.exe — no shell involved
const nodeDir = path.dirname(process.execPath);
const npxCli = path.join(nodeDir, 'node_modules', 'npm', 'bin', 'npx-cli.js');

console.log('node:', process.execPath);
console.log('npx-cli:', npxCli);
console.log('Deploying...');

execFileSync(process.execPath, [
    npxCli, '@insforge/cli', 'deployments', 'deploy', '--env', envString, '.'
], { stdio: 'inherit', cwd: __dirname });

console.log('Done!');
